internal_ws/ykcompile/src/lib.rs

```diff
@@ -1562,7 +1572,7 @@ impl TraceCompiler {
                 let ptr = buf.ptr(dynasmrt::AssemblyOffset(0)) as *mut libc::c_void;
                 let len = buf.len();
                 let alignment = ptr as usize % libc::sysconf(libc::_SC_PAGESIZE) as usize;
-                let ptr = ptr.offset(-(alignment as isize));
+                let ptr = ptr.offset(-(alignment as isize)); // probably better as ptr.pub?
```

Comment:
I assume you mean `ptr.sub`.

Reply:
`<*mut T>::sub` uses pointer arithmetic. Aka it offsets in multiples of `size_of::<T>()`.

Reply:
s/pub/sub/ indeed :) But, as @bjorn3 points out, there are different subtraction functions depending on whether you want bytes or object subtraction.

Reply:
Fixed in db5a334.